apps/twap/src/app/utils/processConditionalOrders.ts

```diff
@@ -0,0 +1,154 @@
+      const orderParts = await generateOrderParts(
+        orderId,
+        orderStruct,
+        blockTimestamp,
+        network.chainId,
+        owner
+      );
+      const partIds = orderParts.map((part) => part.uid);
+
+      // Current execution info (how much is filled etc.) for the TWAP order.
+      const { executedBuyAmount, executedSellAmount } =
+        await orderbook.getExecutionInfo(orderId, partIds);
+
+      const status = getOrderStatus(
+        executedSellAmount,
+        orderStruct,
+        blockTimestamp,
+        singleOrder
+      );
+
+      // Create a transaction, to ensure we don't have any half processed events.
+      fastify.orm.manager.transaction(async (manager) => {
+        // Save block to be able to pick up from here on next run.
+        const block = manager.create(Block, {
+          blockNumber: event.blockNumber,
+          chainId: network.chainId,
+        });
+
+        await manager.save(block);
+
+        // Wallet is Safe address that owns the TWAP order.
+        let wallet = await walletRepository.findOne({
+          where: {
+            address: owner,
+          },
+        });
+
+        if (wallet === null) {
+          wallet = manager.create(Wallet, {
+            address: owner,
+          });
+
+          await manager.save(wallet);
+        }
+
+        const order = new Order();
+
+        order.id = orderId;
+        order.sellToken = orderStruct.sellToken;
+        order.buyToken = orderStruct.buyToken;
+        order.appData = orderStruct.appData;
+        order.receiver = orderStruct.receiver;
+        order.chainId = network.chainId;
+        order.partSellAmount = orderStruct.partSellAmount;
+        order.minPartLimit = orderStruct.minPartLimit;
+        order.t0 = orderStruct.t0;
+        order.n = orderStruct.n;
+        order.t = orderStruct.t;
+        order.span = orderStruct.span;
+        order.status = status;
+        order.wallet = wallet;
+        order.executedBuyAmount = executedBuyAmount;
+        order.executedSellAmount = executedSellAmount;
+
+        // Order parts are the individual orders that make up the TWAP order.
+        const parts = partIds.map((partId) => {
+          const orderPart = new OrderPart();
+          orderPart.id = partId;
+          orderPart.order = order;
+
+          return orderPart;
+        });
+
+        order.parts = parts;
+
+        // Save order and parts (update if it exists)
+        await manager.upsert(Order, order, {
+          conflictPaths: ['id'],
+          upsertType: 'on-duplicate-key-update',
+        });
+      });
+    } catch (err) {
+      // An error has happened, skip event and log it.
+      console.log(
+        `Skipping:\n\tOwner: ${
+          event.args.owner
+        }\n\tOrder ID: ${getConditionalOrderId(
+          event.args.params
+        )}\n\tTx Hash: ${event.transactionHash}\n\tError: ${err.message}`
+      );
+    }
+  }
```

Comment:
i would thing this method is a bit big. Consider simplifying it

Reply:
Good point, will do tomorrow.

Reply:
Simplified this method as much as I could (details: https://github.com/cowprotocol/cow-web-services/pull/8#discussion_r1268739815).

Essentially what I could do was;
* Extract fetching of blockInfo into its own plugin
* Extract fetching of events into a separate function under `conditionalOrders`
* Transformed `processConditionalOrders` to a plugin. This way, we have `fastify` instance given to us as opposed to having it provided every time it is called.

Remaining parts are now scaffolding and massaging the results into a transaction. I don't think we'd have any more benefits from reducing size here, as we'd end up with a lot of parameter passing.